packages/twenty-server/src/engine/core-modules/billing/billing.controller.ts

```diff
@@ -63,6 +63,8 @@ export class BillingController {
     } catch (error) {
       if (error instanceof BillingException) {
         res.status(404).end();
+      } else {
+        res.status(500).end();
```

Comment:
I'm not sure about this try catch:
- agree with the logic that if we don't know the error we should throw a 500
- however ideally we should try to know as much as we can the errors (in your PR description, you mention a timeout on stripe end => ok so if this happens what behavior do we want? we probably want to detect it at the right place in the code, emit a BillingException.StripeAPITimeout and then handle it here to return a 400 rather than a 500 or a 404

Not a blocker for this PR, I think returning a 500 is still an improvement :p